protostar/commands/format/format_command.py

```diff
@@ -0,0 +1,147 @@
+                default=False,
+            ),
+            Command.Argument(
+                name="ignore-broken",
+                description=("Ignore broken files."),
+                type="bool",
+                is_required=False,
+                default=False,
+            ),
+        ]
+
+    async def run(self, args):
+        try:
+            summary, any_unformatted_or_broken = self.format(
+                args.target, args.check, args.log_formatted, args.ignore_broken
+            )
+        except BaseException as exc:
+            self._logger.fatal("Command failed.")
+            raise exc
+        summary.log_summary(log_color_provider)
+
+        # set exit code to 1
+        if any_unformatted_or_broken:
+            raise ProtostarExceptionSilent(
+                "Some files were unformatted, impossible to format or broken."
+            )
+
+    # pylint: disable=too-many-locals
+    def format(
+        self, targets: List[str], check=False, log_formatted=False, ignore_broken=False
+    ) -> Tuple[FormatingSummary, int]:
+        summary = FormatingSummary(self._logger, check, log_formatted)
+        filepaths: List[Path] = []
+
+        for target in targets:
+            target_path = Path(target)
+
+            if target_path.is_file():
+                filepaths.append(target_path.resolve())
+            else:
+                filepaths.extend(
+                    [f for f in target_path.resolve().glob("**/*.cairo") if f.is_file()]
+                )
+
+        any_unformatted_or_broken = False
+
+        for filepath in filepaths:
+            relative_filepath = filepath.relative_to(self._project_root_path)
+
+            try:
+                with open(filepath, "r", encoding="utf-8") as file:
+                    content = file.read()
+                new_content = parse_file(content, str(filepath)).format()
+            except (ParserError, FormattingError) as ex:
+                if ignore_broken:
+                    continue
+
+                summary.extend_and_log(
+                    BrokenFormattingResult(relative_filepath, ex), log_color_provider
+                )
+                any_unformatted_or_broken = True
+
+                # Cairo formatter fixes some broken files
+                # We want to disable this behaviour
+                continue
+
+            if content == new_content:
+                summary.extend_and_log(
+                    CorrectFormattingResult(relative_filepath), log_color_provider
+                )
+            else:
+                if check:
+                    any_unformatted_or_broken = True
+                else:
+                    with open(filepath, "w", encoding="utf-8") as file:
+                        file.write(new_content)
+
+                summary.extend_and_log(
+                    IncorrectFormattingResult(relative_filepath), log_color_provider
+                )
+
+        return summary, any_unformatted_or_broken
```

Comment:
Please move it to a class in `protostar.formatter`.
Split it on smaller functions — Rule of thumb: functions should operate on one level of abstraction.